src/worker/ingestion/process-event.ts

```diff
@@ -268,14 +288,15 @@ export class EventsProcessor {
 
         // CH still needs to update if there are $increment props but Postgres has already done so
         if (arePersonsEqualExcludingIncrement && (!this.db.kafkaProducer || !areTherePropsToIncrement)) {
-            return personFound
+            return returnedProps
         }
 
         if (incrementedPropertiesQueryResult && incrementedPropertiesQueryResult.rows.length > 0) {
             updatedProperties = { ...updatedProperties, ...incrementedPropertiesQueryResult.rows[0].properties }
         }
 
-        return await this.db.updatePerson(personFound, { properties: updatedProperties })
+        await this.db.updatePerson(personFound, { properties: updatedProperties })
```

Comment:
If I read this right, then calling `await this.db.updatePerson(personFound, { properties: updatedProperties })` ... will store as the person's properties only the updated properties, overridding all existing properties... unless an `$increment` property set.